Assets/Scripts/Control&Input/Peer2PeerTransport.cs

```diff
@@ -254,10 +306,17 @@ public static void StartTrainingMode()
     private static IEnumerator WaitAndSwitchToTrainingMode()
     {
         LoadingScreen.Singleton.Show();
+
+        // Wait for player details to be populated
         while (!NetworkClient.isConnected && !singleton.isNetworkActive && players.Count < PlayerInputManagerController.Singleton.LocalPlayerInputs.Count)
             yield return new WaitForEndOfFrame();
         singleton.ServerChangeScene(Scenes.TrainingMode);
-        // TODO until after spawned
+
+        yield return new WaitForSeconds(LoadingScreen.Singleton.MandatoryDuration);
+
+        // Wait for player(s) to have spawned
+        while (FindObjectsByType<PlayerManager>(FindObjectsSortMode.None).Count() < players.Count)
+            yield return new WaitForEndOfFrame();
```

Comment:
Hmmm, is this safe if something wrong happens during initialization? Maybe we need to put in a timeout after x seconds? Probably not too bad if it's only for trainingmode, but still

Reply:
We do need a timeout for this and for the other waiting-for-spawn loops in this class and MatchController. The game *will* just appear to not load if there was an error while spawning players that caused less of them to be spawned.
I just consider this to be less of a problem than the other things that this PR fixes. See #675 (created right now)